parser-typechecker/src/Unison/Codebase/SqliteCodebase/Conversions.hs

```diff
@@ -424,40 +424,30 @@ causalbranch2to1 :: Monad m => (V2.Reference -> m CT.ConstructorType) -> V2.Bran
 causalbranch2to1 lookupCT = fmap V1.Branch.Branch . causalbranch2to1' lookupCT
 
 causalbranch2to1' :: Monad m => (V2.Reference -> m CT.ConstructorType) -> V2.Branch.CausalBranch m -> m (V1.Branch.UnwrappedBranch m)
-causalbranch2to1' lookupCT (V2.Causal hc _he (Map.toList -> parents) me) = do
+causalbranch2to1' lookupCT (V2.Causal hc eh (Map.toList -> parents) me) = do
   let currentHash = causalHash2to1 hc
+      branchHash = branchHash2to1 eh
   case parents of
-    [] -> V1.Causal.UnsafeOne currentHash <$> (me >>= branch2to1 lookupCT)
+    [] -> V1.Causal.UnsafeOne currentHash branchHash <$> (me >>= branch2to1 lookupCT)
     [(hp, mp)] -> do
       let parentHash = causalHash2to1 hp
-      V1.Causal.UnsafeCons currentHash
+      V1.Causal.UnsafeCons currentHash branchHash
         <$> (me >>= branch2to1 lookupCT)
         <*> pure (parentHash, causalbranch2to1' lookupCT =<< mp)
     merge -> do
       let tailsList = map (bimap causalHash2to1 (causalbranch2to1' lookupCT =<<)) merge
       e <- me
-      V1.Causal.UnsafeMerge currentHash <$> branch2to1 lookupCT e <*> pure (Map.fromList tailsList)
+      V1.Causal.UnsafeMerge currentHash branchHash <$> branch2to1 lookupCT e <*> pure (Map.fromList tailsList)
 
 causalbranch1to2 :: forall m. Monad m => V1.Branch.Branch m -> V2.Branch.CausalBranch m
-causalbranch1to2 (V1.Branch.Branch c) = causal1to2' hash1to2cb hash1to2c branch1to2 c
+causalbranch1to2 (V1.Branch.Branch c) =
```

Comment:
This function is where we were previously incorrectly using the causal hash as the branch hash.